adal/authentication_context.py

```diff
@@ -67,8 +70,20 @@ def __init__(self, authority, validate_authority=None, cache=None):
         self._oauth2client = None
         self.correlation_id = None
         env_value = os.environ.get('ADAL_PYTHON_SSL_NO_VERIFY')
+        if api_version is not None:
+            warnings.warn(
+                """The default behavior of including api-version=1.0 on the wire
+                is now deprecated.
+                Future version of ADAL will change the default value to None.

```

Comment:
I have to admit that there is no way to convey the complicated history into this several sentences long warning message.  :-( Regardless of the wording, the main purpose of such warning message was trying to say "hey `api_version=None` is the future, you'd better opt-in (and then test out the new behaviors) now, rather than facing a sudden change in a future major version update".